File: haskell/allergies/haskell/allergies/Allergies.hs
module Allergies (Allergen(..), isAllergicTo, allergies) where

import Data.Bits (testBit)

data Allergen = Eggs | Peanuts | Shellfish | Strawberries | Tomatoes | Chocolate | Pollen | Cats
	deriving (Enum, Show, Eq, Bounded)

allergies :: Int -> [Allergen]
allergies code = filter (`isAllergicTo` code) [minBound ..]

isAllergicTo :: Allergen -> Int -> Bool
isAllergicTo allergen code = testBit code $ fromEnum allergen

File: r/allergies/allergies.R
allergens <- c("eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats")

allergy <- function(num) {
  allergens[as.logical(intToBits(num))] |> na.exclude()
}

allergic_to <- function(allergy_object, allergy) {
  allergy %in% allergy_object
}

list_allergies <- function(allergy_object) {
  allergy_object
}
